csi.proto

```diff
@@ -947,6 +947,26 @@ message GetCapacityResponse {
   // storage. This field is REQUIRED.
   // The value of this field MUST NOT be negative.
   int64 available_capacity = 1;
+
+  // The largest size that could have been used in a CreateVolume call
+  // to create a volume with the same parameters as those in
+  // GetCapacityRequest.
+  //
+  // If `volume_capabilities` or `parameters` is
+  // specified in the request, the Plugin SHALL take those into
+  // consideration when calculating the maximum volume size of the
+  // storage.
+  //
+  // This field is OPTIONAL. If a Plugin has no restrictions for the
+  // size of volumes (for example, linear storage with no fragmentation
+  // issues and no technical limitation for individual volumes), then
+  // can leave this field unset and only return available_capacity.
```

Comment:
Given that we've defined available_capacity very vaguely, and a CO will probably ignore that field, I think a better recommendation here would be to not leave the field blank, but the return the full available capacity, even if that means both fields of this struct have the same value.

Since we don't have a capability bit to go with this feature, I think the presence/absence of this field tells the CO if the SP supports this feature, and therefore we never want to leave it blank, unless the SP doesn't support the feature at all.

Reply:
> Given that we've defined available_capacity very vaguely, and a CO will probably ignore that field,

No, Kubernetes won't ignore it if that's only field that is set. Some vague value is still better than none.

> I think a better recommendation here would be to not leave the field blank, but the return the full available capacity,

But then we need to define what "full capacity" is, right?

We can say "return the same vague value as in `available_capacity`", but benefit does that provide given that a CO must be prepared to fall back to using `available_capacity` when `maximum_volume_size` is unset?

Reply:
No I think we just document that this field should always be the largest possible volume size, and it should never be blank, unless the SP don't support the feature. If the largest possible volume size and the available capacity are the same, that's fine. This allows COs that are only interested in whether a given volume will fit on a given node to answer that question by looking at a single field. The value of the other field could be for sorting the list of possible candidate nodes, if the CO was trying to spread out space consumption, for example.

Reply:
Perhaps I should just leave out the sentence with "If a Plugin has no..."? That the field can be unset is already covered by "is OPTIONAL".

We could say "A Plugin SHOULD set this value to the available capacity if it has no additional limitations (for example, fragmentation) that lead to a smaller maximum volume size."

Reply:
I say keep it simple and stick to what the field means and how it will be interpreted by COs. We should not speculate about the reasons why volume size might be limited to a value smaller than the available capacity in the spec. Just say it means the largest volume size which should be successfully created. COs MAY use this value to make decisions about where to schedule volumes,.

Reply:
Makes sense. Updated.